code/modules/mob/living/carbon/human/human.dm

```diff
@@ -260,13 +260,18 @@
 	popup.open()
 
 // called when something steps onto a human
-// this could be made more general, but for now just handle mulebot
 /mob/living/carbon/human/Crossed(atom/movable/AM)
 	var/obj/machinery/bot/mulebot/MB = AM
 	if(istype(MB))
 		MB.RunOver(src)
 
-	spreadFire(AM)
+	var/mob/living/carbon/M = AM
+	if(M.fire_stacks > 0) //stand on top of a burning person when covered in flammable stuff, catch fire
+		spreadFire(AM)
+		return
+
+	if(do_after(AM, 20, target = null)) //stand still for too long on top of a burning person, catch fire

```

Comment:
This only checks whether the mob that crossed the human is standing still. This means that if the human on fire is moved away from AM after the initial crossing then AM will still catch fire despite not being on the same turf as the human anymore.